index.d.ts

```diff
@@ -52,15 +43,18 @@ export interface AnyAction extends Action {
  *
  * *Do not put API calls into reducers.*
  *
- * @template S State object type.
+ * @template S The type of state consumed and produced by this reducer.
+ * @template A The type of actions the reducer can potentially respond to.
  */
-export type Reducer<S> = (state: S, action: AnyAction) => S;
+export type Reducer<S = {}, A extends Action = Action> = (state: S, action: A) => S;
```

Comment:
Is there a particular reason for `S` to default to `{}`?

Also, I don't think that it is correct to constraint the type of `action`. A reducer must be able to accept every possible action, which is *any object with `type` property*. It should just ignore all actions that don't affect its state.